palisade-service/src/test/java/uk/gov/gchq/palisade/service/palisade/request/AuditErrorMessageTest.java

```diff
@@ -0,0 +1,92 @@
+     *                     This equates to a failure to serialise the string.
+     */
+    @Test
+    public void testSerialiseAuditErrorMessageToJson() throws IOException {
+        Context context = new Context().purpose("testContext");
+        String now = ZonedDateTime.now(ZoneOffset.UTC).format(DateTimeFormatter.ISO_INSTANT);
+
+        AuditErrorMessage auditErrorMessage = AuditErrorMessage.Builder.create()
+                .withUserId("originalUserID")
+                .withResourceId("testResourceId")
+                .withContext(context)
+                .withAttributes(new HashMap<String, Object>())
+                .withError(new InternalError("Something went wrong!"));
+
+        JsonContent<AuditErrorMessage> auditErrorMessageJsonContent = jsonTester.write(auditErrorMessage);
+
+        assertThat(auditErrorMessageJsonContent).extractingJsonPathStringValue("$.userId").isEqualTo("originalUserID");
+        assertThat(auditErrorMessageJsonContent).extractingJsonPathStringValue("$.resourceId").isEqualTo("testResourceId");
+        assertThat(auditErrorMessageJsonContent).extractingJsonPathStringValue("$.context.contents.purpose").isEqualTo("testContext");
+        assertThat(auditErrorMessageJsonContent).extractingJsonPathStringValue("$.error.message").isEqualTo("Something went wrong!");
+    }
+
+    /**
+     * Create the object from a Json string and then test the content of the object.
+     *
+     * @throws IOException if it fails to parse the object
+     */
+    @Test
+    public void testDeserialiseJsonToAuditErrorMessage() throws IOException {
+        String jsonString = "{\"userId\":\"originalUserID\",\"resourceId\":\"testResourceId\",\"context\":{\"class\":\"uk.gov.gchq.palisade.Context\",\"contents\":" +
+                "{\"purpose\":\"testContext\"}},\"error\":{\"cause\":null,\"stackTrace\":[],\"message\":\"Something went wrong!\",\"suppressed\":[],\"localizedMessage" +
+                "\":\"Something went wrong!\"},\"serviceName\":\"palisade-service\",\"timestamp\":\"2020-01-01T08:00:00.000000Z\",\"serverIP" +
+                "\":\"192.168.1.1\",\"serverHostname\":\"host.name\",\"attributes\":{}}";
+        ObjectContent<AuditErrorMessage> auditSuccessMessageObjectContent = jsonTester.parse(jsonString);
+
+        AuditErrorMessage auditErrorMessage = auditSuccessMessageObjectContent.getObject();
+        assertThat(auditErrorMessage.getUserId()).isEqualTo("originalUserID");
+        assertThat(auditErrorMessage.getResourceId()).isEqualTo("testResourceId");
+        assertThat(auditErrorMessage.getContext().getPurpose()).isEqualTo("testContext");
+        assertThat(auditErrorMessage.getServiceName()).isEqualTo("palisade-service");
+        assertThat(auditErrorMessage.getServerIP()).isEqualTo("192.168.1.1");
+        assertThat(auditErrorMessage.getServeHostName()).isEqualTo("host.name");
+        assertThat(auditErrorMessage.getError().getMessage()).isEqualTo("Something went wrong!");
```

Comment:
check that the time gets set correctly

Reply:
one issue with the time is that it audits using system time, to the millisecond, then when it compares it to "now" that time has advanced, even if its by 800ms, it'll therefore fail the compairson